### ISSUE TITLE
Makefiles or npm scripts?

### ISSUE THREAD

**Issue by charypar** [jameshopkins/webapptest#44]
Place to discuss approach to running tasks.

My 2c: npm scripts are cross platform and can use locally installed dependencies with ease. Looking at what we're doing we don't really benefit from `make`s ability to only rebuild what changed. Not sure of the merit of using it in this case.

I don't really feel very strongly about this though.

**Comment by jameshopkins** [comment]
> Looking at what we're doing we don't really benefit from makes ability to only rebuild what changed

Yep. Right now, there is no functional merit to using `npm` scripts over `make` or vice-versa.

We initially started out with using just `npm` scripts, since we're only dealing with JS-based tooling in a Node environment. However, I don't think we should get too invested in the JS ecosystem especially this early into the project. After getting past the MVP, we should definitely evaluate [potentially better tech for certain jobs](https://github.com/jameshopkins/webapptest/issues/42), and a build tool that's agnostic to the technology in which the project is made up of, will be beneficial potentially.

For example, it wouldn't necessarily make sense to register scripts in `/app`'s `package.json` since it's there as a side-effect of both `/app/ui` and `/app/server` consuming it's dependencies.

**Comment by lpil** [comment]
I think that make is nicer to use, even if we don't use any of it's extra features. :)

**Comment by lpil** [comment]
> Right now, there is no functional merit to using npm scripts over make or vice-versa.

Actually we could use `-j` to speed up webpack builds on multicore machines. :)

**Comment by jameshopkins** [comment]
## Proposal

If one of the application's internal dependencies (i.e `/app/server`, `/app/ui`, `/agent/server`, using Node - and therefore NPM) uses a technology that allows for the ability to register shell scripts, then use that interface. `make` resides at the root, `/app` and `/agent` level and is responsible for targeting the relevant internal dependency, and invoking the registered scripts.

**Comment by jameshopkins** [comment]
> Actually we could use -j to speed up webpack builds on multicore machines. :)

Nice 👍
